pkg/kubelet/kubelet.go

```diff
@@ -2082,6 +2089,12 @@ func (kl *Kubelet) SyncTerminatedPod(ctx context.Context, pod *v1.Pod, podStatus
 	}
 	klog.V(4).InfoS("Pod termination unmounted volumes", "pod", klog.KObj(pod), "podUID", pod.UID)
 
+	// TODO: unsure the pod volumes are removed
+	if kl.podVolumesExist(pod.UID) && !kl.keepTerminatedPodVolumes {
```

Comment:
this should be a wait loop:

```
if !kl.keepTerminatedPodVolumes {
   if err := wait.PollUntilContextCancel(..., func(...) (bool, error) {    return kl.podVolumesExist(pod.UID), nil  }); err != nil {
    return err
  }
}
```

i think

Reply:
Pushed some initial implementation. I think it should be delegated to volumeManager where the interval is defined as a const. Going to follow up.